packages/marshal/src/marshal.js

```diff
@@ -866,12 +866,19 @@ export function makeMarshal(
               X`invalid error message typeof ${q(typeof message)}`,
             );
             const EC = getErrorConstructor(`${name}`) || Error;
-            const error = harden(new EC(`${message}`));
-            ibidTable.register(error);
+            let error;
             if (typeof errorId === 'string') {
               // errorId is a late addition so be tolerant of its absence.
+              error = assert.error(
+                `${message}`,
+                EC,
+                `Remote${EC.name}(${errorId})`,
```

Comment:
does the inclusion of the third `assert.error()` argument remove the need for the following `assert.note()` call?

Reply:
Good point. And killing it led to further code simplification.

Reply:
Argghh. By making this simplification before https://github.com/endojs/endo/pull/649 is released, and before agoric-sdk depends on that release, we're losing the "Received as" note but not yet seeing the information in the detailed error name. I am proceeding with this simplification anyway, raising the priority of another ses release with https://github.com/endojs/endo/pull/649 . Attn @kriskowal @dckc .